zeitschrift-fur-kunstgeschichte.csl

```diff
@@ -149,14 +159,29 @@
                     </group>
                     <names variable="editor translator" delimiter=", ">
                       <label form="verb-short" suffix=" "/>
-                      <name/>
+                      <name and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never"/>
                     </names>
                   </if>
+                  <else-if match="any" variable="translator">
+                    <choose>
+                      <if match="none" variable="author">
+                        <names variable="editor">
+                          <name and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never"/>
+                          <label form="short" prefix=" (" suffix=")"/>
+                        </names>
+                      </if>
+                    </choose>
+                    <group delimiter=" ">
+                      <text variable="title" font-style="italic" suffix=","/>
+                      <text macro="translator"/>
+                      <text macro="collection-with-number"/>
+                    </group>
+                  </else-if>
```

Comment:
IMO you don't need here another `else-if`-case for the translator(s). Try to use another group, i.e. something like:
```
                   <else>
                     <choose>
                       <if match="none" variable="author">
                         <names variable="editor">
                           <name/>
                           <name and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never"/>
                           <label form="short" prefix=" (" suffix=")"/>
                         </names>
                       </if>
                     </choose>
                     <group delimiter=" ">
                       <group delimiter=", ">
                         <text variable="title" font-style="italic"/>
                         <text macro="translator"/>
                       </group>
                       <text macro="collection-with-number"/>
                     </group>
                   </else>
```